### ISSUE TITLE
Consider adding a "gpuadapterschanged" event to the Web platform

### ISSUE THREAD

**Issue by kainino0x** [gpuweb/gpuweb#221]
This event is becoming more important with external GPUs becoming more common. It would probably live on navigator.gpu.ongpuadapterschanged.

It would be useful for both WebGPU and WebGL users - it probably shouldn't actually be tied to the WebGPU API (I think).

**Comment by Kangz** [comment]
@kainino0x is this still needed after the many reworks of `requestAdapter/requestDevice`?

**Comment by kainino0x** [comment]
Yes, we still don't have a way for apps to detect newly available adapters.
